stan/math/prim/mat/fun/multiply.hpp

```diff
@@ -66,13 +67,27 @@ inline Eigen::Matrix<double, R1, C2> multiply(
  * @return Scalar result of multiplying row vector by column vector.
  * @throw std::domain_error if rv and v are not the same size.
  */
-template <int C1, int R2>
-inline double multiply(const Eigen::Matrix<double, 1, C1>& rv,
-                       const Eigen::Matrix<double, R2, 1>& v) {
+template <int C1, int R2, typename T1, typename T2,
+          typename = enable_if_all_arithmetic<T1, T2>>
+inline return_type_t<T1, T2> multiply(const Eigen::Matrix<T1, 1, C1>& rv,
+                                      const Eigen::Matrix<T2, R2, 1>& v) {
   check_matching_sizes("multiply", "rv", rv, "v", v);
   return rv.dot(v);
 }
 
+/**
+ * Return specified matrix multiplied by specified scalar.
+ * @tparam R Row type for matrix.
+ * @tparam C Column type for matrix.
+ * @param m Matrix.
+ * @param c Scalar.
+ * @return Product of matrix and scalar.
+ */
+template <typename T1, typename T2, typename = enable_if_all_arithmetic<T1, T2>>
+inline return_type_t<T1, T2> multiply(T1 m, T2 c) {
+  return c * m;
+}
```

Comment:
Not sure if this was intended, but we never actually had a `multiply` for two arithmetic types? I'm pretty sure calling 

```cpp
double foo = multiply(1.0, 1.0);
```

Would actually call the version in `rev`?